functions/model/tasks/tark/editTask.js

```diff
@@ -197,6 +197,10 @@ exports.editTask = function(request, response) {
             promises.push(updateSprintPromise);
         }
 
+        if(milestoneId == undefined) {
+            milestoneId=""
+        }
+
```

Comment:
The best practice would be to handle this milestone issue from UI, better not to send undefined values to backend.